### PR TITLE
Make sync field an object indexed by id

### PR DESCRIPTION
**Pull request by Crash--**
We try to follow as can as possible this spec for the next version of our Contacts App. Since 1 contact can be synchronized with multiple sources, we need to change the shape of this object. 

What do you think about this change?

### CODE REVIEW

File: packages/cozy-konnector-libs/docs/synchronization.md [hunk 1]
```diff
@@ -67,10 +67,16 @@ To store synchronization data, we are using a `sync` attribute in document `meta
...
   "metadata": {
     "sync": {
-      "id": "7ee401e841c94159addb47f190903139",
-      "konnector": "trainline",
-      "last_sync": "Mon, 12 Feb 2018 16:25:34 GMT"
-    }
+      "7ee401e841c94159addb47f190903139": {
+        "id": "7ee401e841c94159addb47f190903139",
+        "konnector": "trainline",
+        "last_sync": "Mon, 12 Feb 2018 16:25:34 GMT"
```
Comment (ptbrowne) [inline packages/cozy-konnector-libs/docs/synchronization.md]: Why isn't this a ISO date ?

**Comment by ptbrowne** [comment]
Fine by me but I haven't touched to synchronization much. I'd like to also have the input of @doubleface and @nono .

File: packages/cozy-konnector-libs/docs/synchronization.md [hunk 2]
```diff
@@ -67,10 +67,16 @@ To store synchronization data, we are using a `sync` attribute in document `meta
 {
   "metadata": {
     "sync": {
-      "id": "7ee401e841c94159addb47f190903139",
-      "konnector": "trainline",
-      "last_sync": "Mon, 12 Feb 2018 16:25:34 GMT"
-    }
+      "7ee401e841c94159addb47f190903139": {
+        "id": "7ee401e841c94159addb47f190903139",
```
Comment (nono) [inline packages/cozy-konnector-libs/docs/synchronization.md]: Why `7ee401e841c94159addb47f190903139` is repeated? First as the key, and then as an id.

File: packages/cozy-konnector-libs/docs/synchronization.md (hunk 1 above)
Comment (Crash--) [reply packages/cozy-konnector-libs/docs/synchronization.md]: As the co-writer of this spec, tell me! :D 

Being serious, you'll use the ISO format `2018-10-19T10:58:37.025688+02:00` if @gregorylegarec has no objection ;)

**Comment by nono** [comment]
Do you aim synchronization in only one direction (import from an external service to cozy)?

**Comment by nono** [comment]
I'm also curious to know why we don't use the `importedAt` / `importedFrom` described in https://github.com/cozy/cozy-doctypes/tree/master/docs#document-metadata

File: packages/cozy-konnector-libs/docs/synchronization.md (hunk 2 above)
Comment (nono) [reply packages/cozy-konnector-libs/docs/synchronization.md]: OK, if it is more convenient, it's OK for me.

**Comment by Crash--** [comment]
Understood. Let me finish our spec and I'll update this document with all the needed modifications. I just wanted to have an input for the proposed change at the beginning. 

But to answer your question @nono : I can have a contact importedFrom: Google-Konnector, but also synched with iCloud.

**Comment by nono** [comment]
OK, you can still do a `s/metadata/cozyMetadata/`

**Comment by nono** [comment]
Ping @Crash--

**Comment by Crash--** [comment]
@doubleface If I read from https://github.com/cozy/cozy-doctypes/blob/master/docs/io.cozy.contacts.md it seems that sync is into `cozyMetadata`. But here, it's in `metadata`. Do you know the final word?

**Pull request closed without merging**